src/utils/googleTagManager.ts

```diff
@@ -155,14 +166,63 @@ export const trackEvent = ({
   action: string
   label?: EventLabel
 }): void => {
-  const dataLayer = {
+  const dataLayer: EventDataLayer = {
     event,
     chainId: _getChainId(),
     eventCategory: category,
     eventAction: action,
     eventLabel: tryParse(label),
   }
 
+  track(dataLayer)
+}
+
+const ETHEREUM_ADDRESS_REGEX = /0x[a-fA-F0-9]{40}/g
```

Comment:
I would assume we already have something like this in the project somewhere. Have you checked this?